src/spatial_reaction_systems/spatial_reactions.jl

```diff
@@ -0,0 +1,96 @@
+isedgeparameter(x::Num, args...) = isedgeparameter(Symbolics.unwrap(x), args...)
+function isedgeparameter(x, default = false)
+    p = Symbolics.getparent(x, nothing)
+    p === nothing || (x = p)
+    Symbolics.getmetadata(x, EdgeParameter, default)
+end
+
+### Transport Reaction Structures ###
+
+# A transport reaction. These are simple to hanlde, and should cover most types of spatial reactions.
+# Only permit constant rates (possibly consisting of several parameters).
+struct TransportReaction <: AbstractSpatialReaction
+    """The rate function (excluding mass action terms). Currently only constants supported"""
+    rate::Num
+    """The species that is subject to difusion."""
+    species::BasicSymbolic{Real}
+
+    # Creates a diffusion reaction.
+    function TransportReaction(rate::Num, species::Num)
+        new(rate, species.val)
+    end
+end
+# If the rate is a value, covert that to a numemric expression.
+function TransportReaction(rate::Number, args...)
+    TransportReaction(Num(rate), args...)
+end
+# Creates a vector of TransportReactions.
+function transport_reactions(transport_reactions)
+    [TransportReaction(tr[1], tr[2]) for tr in transport_reactions]
+end
+
+# Macro for creating a transport reaction.
+macro transport_reaction(rateex::ExprValues, species::Symbol)
+    make_transport_reaction(MacroTools.striplines(rateex), species)
+end
+function make_transport_reaction(rateex, species)
+    parameters = [];
+    find_parameters_in_rate!(parameters, rateex)
+    quote
+        $(isempty(parameters) ? nothing : :(@parameters $(parameters...)))
+        @variables t
+        @species $(species)(t)
+        TransportReaction($rateex, $species)
+    end
+end
+
+# Gets the parameters in a transport reaction.
+ModelingToolkit.parameters(tr::TransportReaction) = convert(Vector{BasicSymbolic{Real}}, Symbolics.get_variables(tr.rate))
+
+# Gets the species in a transport reaction.
+species(tr::TransportReaction) = [tr.species]
+spatial_species(tr::TransportReaction) = [tr.species]
+
+# Checks that a transport reaction is valid for a given reaction system.
+function check_spatial_reaction_validity(rs::ReactionSystem, tr::TransportReaction; edge_parameters=[])
+    # Checks that the species exist in the reaction system (ODE simulation code becomes difficult if this is not required, as non-spatial jacobian and f function generated from rs is of wrong size).  
+    any(isequal(tr.species), species(rs)) || error("Currently, species used in TransportReactions must also be in non-spatial ReactionSystem. This is not the case for $(tr.species).")
+    # Checks that the rate does not depend on species.    
+    isempty(intersect(ModelingToolkit.getname.(species(rs)), ModelingToolkit.getname.(Symbolics.get_variables(tr.rate)))) || error("The following species were used in rates of a transport reactions: $(setdiff(ModelingToolkit.getname.(species(rs)), ModelingToolkit.getname.(Symbolics.get_variables(tr.rate)))).")
```

Comment:
Everything in the transportation reaction rate is parameters, by default. We just want to ensure that their names are not reused, since this would be bad.